### DISCUSSION TITLE
🚀 Feature: Something like Claims in FirebaseAuth

### DISCUSSION THREAD

**Discussion by bw-flagship** [appwrite/appwrite#11510] [Ideas]
### 🔖 Feature description

I come from firebase auth, where I used custom claims a lot.
E.g. if a user can create a purchase, I add a claim "isPremium":"true" to his token, so the backend knows he is premium without querying the database.

I dont think there is such a thing in appwrite, right?

### 🎤 Pitch

Save database load by drastically reduce the amount of necessary queries.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by eldadfux** [comment]
We had some discussion on custom user labels or roles. The idea was to have a way for defining custom labels or roles for each users which potentially could also be used for permission roles. The roles or labels could be only set using the server API or Cloud Function for security. This roles/labels would allow to segment data to only specific groups of users, for example: admins, vips, or premium subscribers.

Does this sound like an elegant solution for you use case?

You can learn more about it on these discussions: 
https://github.com/appwrite/appwrite/discussions/2317
https://github.com/appwrite/appwrite/discussions/2382#discussioncomment-1600838

**Comment by bw-flagship** [comment]
Thanks for the prompt response! Those potential features seem to solve a lot of the problem. Claims have the additional advantage that the user can extract them from their token as well, so there is no need for passing the information via additional requests.
Since claims are not just a Firebase thing, but a very wide-adopted concept in the world of auth-tools, I don't quite understand why you are looking for workarounds instead of just implementing claims?

Thanks for your time!

**Comment by DH-555** [comment]
I think that this has been implemented with labels, right?
